src/Avalonia.Base/Controls/ResourceDictionary.cs

```diff
@@ -147,6 +147,9 @@ public void AddDeferred(object key, Func<IServiceProvider?, object?> factory)
         public void AddDeferred(object key, IDeferredContent deferredContent)
             => Add(key, deferredContent);
 
+        public void AddNotShared(object key, INotSharedDeferredContent notSharedDeferredContent)
```

Comment:
This should take a `IDeferredContent` directly.

We can then wrap it inside the dictionary with a private marker class. A single instance overhead is acceptable here, as we expect most resources to have the default `x:Shared=true`.